### DISCUSSION TITLE
focus-visible instead of focus?

### DISCUSSION THREAD

**Discussion by brockfanning** [uswds/uswds#4915] [Ideas]
Hi all, I just became aware of the [focus-visible](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible) pseudo-class and I was wondering if there had been any discussion of switching over to focus-visible instead of focus. My understanding is that this would prevent the focus outline from appearing on links/buttons when they are mouse-clicked, but keep the current behavior when they are tab-focused; and that this would not change any behavior when it comes to form inputs.

It's sometimes a bit of a struggle to convince clients that the focus outlines are necessary, and I suspect many would like for the focus outlines to be less common for mouse users. But I'm sure there are many considerations so I'm interested in thoughts from the community. Thanks!

**Comment by amyleadem** [comment]
Hi @brockfanning, 
Thanks for sharing this idea! I have opened an [issue](https://github.com/uswds/uswds/issues/4917) from this discussion so that we can get this on the USWDS project board. The core team will review, discuss and update the issue as it develops. If you have any other questions or comments, please feel free to reach out. 

Related issue: https://github.com/uswds/uswds/issues/4917

**Comment by amyleadem** [comment]
Hello @brockfanning,
I wanted to cycle back to let you know that the core team has discussed the topic of integrating `:focus-visible` into USWDS and that we have decided against it at this time. For accessibility reasons, we want to stay with what we know and what is consistent for our users. With `:focus`, we know that users will get a consistent, predictable experience regardless of browser or input method. With `:focus-visible`, the risk of a varied and confusing experience goes up. This is especially true for users that browse with more than one input method. For this reason, we are sticking with `:focus` at this time. 

We appreciate you sharing this idea and want to thank you for thinking about USWDS. Getting suggestions like this from the community are what keep our product great.

If you have any questions or want to continue the discussion, please let us know!
